Nodejs/Product/TypeScriptSourceMapReader/SourceMapReaderException.cs

```diff
@@ -0,0 +1,138 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Microsoft.NodejsTools.TypeScriptSourceMapReader {
+    /// <summary>
+    /// Type of SourceMapException
+    /// </summary>
+    public enum SourceMapReaderExceptionKind {
+        InvalidSourceMapUrlException,
+        SourceMapReadFailedException,
+        UnsupportedFormatSourceMapException,
+        ErrorDecodingSourcemapException
+    }
```

Comment:
Is this needed? I don't see it being used outside of this file. Additional, class type can be checked with `is`.